src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -0,0 +1,590 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Internal.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Class for managing wrappers of COM IUnknown types.
+    /// </summary>
+    public abstract partial class ComWrappers
+    {
+        internal static IntPtr DefaultIUnknownVftblPtr { get; } = CreateDefaultIUnknownVftbl();
+
+        internal static Guid IID_IUnknown = new Guid(0x00000000, 0x0000, 0x0000, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46);
+
+        const int DispatchAlignmentThisPtr = 16;
+        private static readonly ConditionalWeakTable<object, ManagedObjectWrapperHolder> CCWTable = new ConditionalWeakTable<object, ManagedObjectWrapperHolder>();
+
+        /// <summary>
+        /// ABI for function dispatch of a COM interface.
+        /// </summary>
+        public partial struct ComInterfaceDispatch
+        {
```

Comment:
```suggestion
        {
            internal IntPtr Vtable;
            internal ManagedObjectWrapper* thisPtr;
```
from DispatchSectionEntry

Reply:
My understanding that `DispatchSectionEntry` has different memory layout. `ThisPtr` then `Vtable`. Since this is part of ABI it would be breaking change. no?

Reply:
The layout of `DispatchSectionEntry` is not part of the API. It is an implementation detail that is fine to change.

The only part of the API is that "this" pointer is set to `&VTable` when the COM method is called.

Reply:
I have to re-introduce DispatchSectionEntry as 
```
struct DispatchSectionEntry
{
            internal IntPtr Vtable;
            internal ManagedObjectWrapper* thisPtr;
}
```
because `ComInterfaceDispatch` shared with regular runtime, and thus if I add new field here. it will produce following error `error CS0282: There is no defined ordering between fields in multiple declarations of partial struct 'ComWrappers.ComInterfaceDispatch'. To specify an ordering, all instance fields must be in the same declaration.` 

Other option, is to move that field from `ComWrappers.cs` to maybe `ComWrappers.Windows.cs` ?

Reply:
Ok, I think you can keep the separate definition then. I would call it something like `InternalComInterfaceDispatch` to make easier to see that it is the same structure as `ComInterfaceDispatch`.